app/routers/auth.py

```diff
@@ -0,0 +1,72 @@
+from fastapi import APIRouter, Depends, HTTPException, Security
+from fastapi.security import OAuth2PasswordBearer
+from jose import JWTError, jwt
+import requests
+import json
+import base64
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPublicNumbers
+from cryptography.hazmat.primitives import serialization
+
+router = APIRouter()
+
+# Keycloak configuration
+KEYCLOAK_URL = "http://localhost:8080"  # Your Keycloak server address

```

Comment:
Not your concern - but we should maybe more this to main.py or have all the key configuration in the one place. 

Something for next interns to look at.